### PR TITLE
[istio] Exclude upmeter from discovery by istiod

### PR DESCRIPTION
**Pull request by apolovov**
## Description
Excluding d8-related namespaces from istiod discovery.

## Why do we need it, and what problem does it solve?
Fixes https://github.com/deckhouse/deckhouse/issues/1906.

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: istio
type: feature
summary: Excluding d8-related namespaces from istiod discovery.
impact: D8 services won't be accessible from applications (except d8-user-authn and d8-ingress-nginx).
```

### CODE REVIEW

File: ee/modules/110-istio/docs/README_RU.md [hunk 1]
```diff
@@ -286,4 +286,4 @@ Istio позволяет осуществлять сбор трейсов с п
 ## Накладные расходы
 
 [Примерная оценка накладных расходов при использовании Istio.](https://istio.io/latest/docs/ops/deployment/performance-and-scalability/)
-Для ограничения потребления ресурсов путём ограничения поля видимости sidecar, возможно использование ресурса [Sidecar](istio-cr.html#sidecar).
+Для ограничения потребления ресурсов путём ограничения поля видимости sidecar, возможно использование ресурса [Sidecar](istio-cr.html#sidecar), а также предусмотрен лейбл для исключения неймспейса из поля зрения Istio — `istio.deckhouse.io/discovery: disabled`.
```
Comment (z9r5) [inline ee/modules/110-istio/docs/README_RU.md:289]:
```suggestion
Для ограничения потребления ресурсов путём ограничения поля видимости sidecar, возможно использование ресурса [Sidecar](istio-cr.html#sidecar), а также предусмотрен лейбл для исключения пространства имен из поля зрения Istio — `istio.deckhouse.io/discovery: disabled`.
```

**Review by z9r5 — APPROVED** [review]

File: ee/modules/110-istio/docs/README_RU.md (hunk 1 above)
Comment (apolovov) [reply ee/modules/110-istio/docs/README_RU.md]: Уже не актуально(

**Review by shvgn — DISMISSED** [review]
LGTM, 

This doc helped me to understand the discovery rules. Would be pleasant to see the doc link right here in comments

https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig

**Review by kvaps — DISMISSED** [review]
LGTM

File: ee/modules/110-istio/templates/control-plane/iop.yaml [hunk 2]
```diff
@@ -60,6 +60,12 @@ spec:
   meshConfig:
     rootNamespace: d8-{{ $.Chart.Name }}
     trustDomain: {{ $.Values.global.discovery.clusterDomain | quote }}
+    discoverySelectors:
+    - matchExpressions:
```
Comment (nabokihms) [inline ee/modules/110-istio/templates/control-plane/iop.yaml:64]: If I get it right, it means all deckhouse and upmeter namespaces are excluded except istio, ingress-nginx, user-authn. Could you please leave a comment on why we need these three particular namespaces and why we exclude others (and also ones owned by upmeter)?

**Review by yalosev — APPROVED** [review]

File: ee/modules/110-istio/templates/control-plane/iop.yaml (hunk 2 above)
Comment (apolovov) [reply ee/modules/110-istio/templates/control-plane/iop.yaml:72]: Done.

File: ee/modules/110-istio/templates/control-plane/iop.yaml [hunk 3]
```diff
@@ -60,6 +60,23 @@ spec:
...
+    # with enableIstioSidecar and d8-user-authn to interact between applications and the dex installation.
+    discoverySelectors:
+      # Include d8-istio, d8-ingress-nginx and d8-user-authn namespaces:
+    - matchExpressions:
+      - {key: "heritage", operator: In, values: [deckhouse]}
+      - {key: "module",   operator: In, values: [istio,ingress-nginx,user-authn]}
+
+      # Exclude everything else related to deckhouse and upmeter:
+    - matchExpressions:
+      - {key: "heritage", operator: NotIn, values: [deckhouse,upmeter]}
```
Comment (kvaps) [inline ee/modules/110-istio/templates/control-plane/iop.yaml:78]: The operators generated services have no heritage label, I found three in my cluster:

`-n d8-monitoring svc/kubelet`
`-n d8-monitoring svc/prometheus-operated`
`-n d8-linstor svc/linstor`

Comment (kvaps) [reply ee/modules/110-istio/templates/control-plane/iop.yaml:78]: Ah, this is about namespaces, LGTM then

**Review by kvaps — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Pull request merged**